packages/compass-workspaces/src/stores/workspaces.ts

```diff
@@ -455,40 +505,48 @@ export const openWorkspace = (
 > => {
   return (dispatch, getState, { instance, dataService }) => {
     const oldTabs = getState().tabs;
-    if (
-      workspaceOptions.type === 'Collection' &&
-      !getState().collectionInfo[workspaceOptions.namespace]
-    ) {
-      // Fetching extra meta for collection should not block tab opening
-      void (async () => {
-        const { database, collection } = toNS(workspaceOptions.namespace);
-        try {
-          const coll = await instance.getNamespace({
-            dataService,
-            database,
-            collection,
-          });
-          if (coll) {
-            await coll.fetch({ dataService });
-            dispatch({
-              type: WorkspacesActions.FetchCollectionTabInfo,
-              namespace: workspaceOptions.namespace,
-              info: {
-                isTimeSeries: coll.isTimeSeries,
-                isReadonly: coll.readonly ?? coll.isView,
-                sourceName: coll.sourceName,
-              },
+    if (workspaceOptions.type === 'Collection') {
+      if (!getState().collectionInfo[workspaceOptions.namespace]) {
+        // Fetching extra meta for collection should not block tab opening
+        void (async () => {
+          const { database, collection } = toNS(workspaceOptions.namespace);
+          try {
+            const coll = await instance.getNamespace({
+              dataService,
+              database,
+              collection,
             });
+            if (coll) {
+              await coll.fetch({ dataService });
+              dispatch({
+                type: WorkspacesActions.FetchCollectionTabInfo,
+                namespace: workspaceOptions.namespace,
+                info: {
+                  isTimeSeries: coll.isTimeSeries,
+                  isReadonly: coll.readonly ?? coll.isView,
+                  sourceName: coll.sourceName,
+                },
+              });
+            }
+          } catch (err) {
+            // It's okay if we failed to fetch this optional metadata, this error
+            // can be ignored
+            if ((err as Error).name === 'MongoServerError') {
+              return;
+            }
+            throw err;
```

Comment:
Should this really end up being an unhandled rejection? Logging the error (maybe even in both the `MongoServerError` and non-`MongoServerError`) might be a better choice here

Reply:
I am fine with adding that and removing this. Nonetheless, I'd let @gribnoysup comment on this as this was here before.

Reply:
I think we were talking at some point about not swallowing all the errors if we don't expect them, but can't say I remember arriving on a decision, so feel free to remove the throw. Adding logging make sense